solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -1311,6 +1311,22 @@ public Builder withProxyConfiguration(
       this.proxyIsSecure = isSecure;
       return this;
     }
+
+    /**
+     * Setup basic authentication from a string formatted as username:password. If the string is
+     * Null then it doesn't do anything.
+     *
+     * @param credentials The username and password formatted as username:password
+     * @return this Builder
+     */
+    public Builder withOptionalBasicAuthCredentials(String credentials) {
```

Comment:
if there is no check for existence of `:` at least an `assert credentials.contains(":")` would be good

Reply:
```
throw new IllegalStateException(
              "Invalid Authentication credentials. Either both username and password or none must be provided");
```
this appears to be a pattern in other builders...

Reply:
See what I added!